src/RTUMaster.cpp

```diff
@@ -1,75 +1,95 @@
 #include <modbus/RTUMaster.hpp>
 
-#include <modbus/common.hpp>
 #include <modbus/Exceptions.hpp>
 #include <modbus/RTU.hpp>
+#include <modbus/common.hpp>
 
 using namespace std;
 using namespace base;
 using namespace modbus;
 
-int RTUMaster::extractPacket(uint8_t const* buffer, size_t bufferSize) const {
+int RTUMaster::extractPacket(uint8_t const* buffer, size_t bufferSize) const
+{
     throw std::logic_error("modbus::RTUMaster should be read only using readRaw");
 }
 
-RTUMaster::RTUMaster()
-    : iodrivers_base::Driver(RTU::FRAME_MAX_SIZE * 10) {
+RTUMaster::RTUMaster(uint8_t error_threshold, uint8_t error_increment)
+    : iodrivers_base::Driver(RTU::FRAME_MAX_SIZE * 10)
+{
     setReadTimeout(base::Time::fromSeconds(1));
     m_read_buffer.resize(MAX_PACKET_SIZE);
     m_write_buffer.resize(MAX_PACKET_SIZE);
     m_frame.payload.reserve(RTU::FRAME_MAX_SIZE);
+
+    m_error_increment = error_increment;
+    m_error_threshold = error_threshold;
 }
 
-void RTUMaster::setInterframeDelay(base::Time const& delay) {
+void RTUMaster::setInterframeDelay(base::Time const& delay)
+{
     m_interframe_delay = delay;
 }
 
-base::Time RTUMaster::getInterframeDelay() const {
+base::Time RTUMaster::getInterframeDelay() const
+{
     return m_interframe_delay;
 }
 
-Frame RTUMaster::readFrame() {
+void RTUMaster::setErrorThreshold(uint8_t const& threshold)
```

Comment:
```suggestion
void RTUMaster::setErrorThreshold(uint8_t threshold)
```

pass argument as constant reference when they are non-fundamental types